jdatetime/__init__.py

```diff
@@ -645,6 +646,35 @@ def aslocale(self, locale):
 date.max = date(MAXYEAR, 12, 30)
 
 
+if sys.version_info[:2] < (3, 7):
+    # Prior to Python 3.7, `%` is escaped by re.escape. We don't want that.
+    def _escape(pattern):
+        if isinstance(pattern, bytes):  # only happens in Python 2
+            pattern = pattern.decode()
+        return _re.escape(pattern).replace(r'\%', '%')
+else:
+    _escape = _re.escape
```

Comment:
Please move this to the top of the file

Reply:
IMO this seems reversed. Instead of changing the definition of the function based on the python version, the function body could work differently based on python version. for example

```python
dev _pattern_escape(pattern):
    if  sys.version_info[:2] < (3, 7):
        ...
    else:
        ...
```

Reply:
Then, the if condition (`if  sys.version_info[:2] < (3, 7):`) should be executed in every call of `_pattern_escape`.
But with the original version, we just check once and then just use the proper function based on Python version.
I would suggest keeping it as is.